### ISSUE TITLE
Avoiding numerical issues in `show` for problematic chains

### ISSUE THREAD

**Issue by torfjelde** [TuringLang/MCMCChains.jl#410]
#401 is dope stuff (thanks @sethaxen ) _but_ the tail ess in many cases seem to cause errors upon trying upon calling `show`, e.g. I just got the following trying a model

```julia
DivideError: integer division error

Stacktrace:
  [1] div
    @ ./int.jl:288 [inlined]
  [2] divgcd(x::Int64, y::Int64)
    @ Base ./rational.jl:45
  [3] //
    @ ./rational.jl:73 [inlined]
  [4] /(x::Rational{Int64}, y::Rational{Int64})
    @ Base ./rational.jl:346
  [5] _ess_rhat(::Val{:basic}, chains::Array{Real, 3}; autocov_method::AutocovMethod, split_chains::Int64, maxlag::Int64)
    @ MCMCDiagnosticTools ~/.julia/packages/MCMCDiagnosticTools/oVEPy/src/ess_rhat.jl:520
  [6] #_ess#36
...
```

This is for a problematic chain of course (I believe the variance is _very_ high?). I also ran into a bug somewhere earlier, again, in the ESS computation.

Maybe we should provide a way to convert these errors into warnings in `MCMCChains.describe`?

https://github.com/TuringLang/MCMCChains.jl/blob/ddac60fb57b8dba6955922cec46573b310dbc0c1/src/stats.jl#L286-L331

Maybe just add try-catch statements to all of these and do `@warn` + `NaN` in case of failure?

**Comment by devmotion** [comment]
At first glance (haven't checked anything), it seems we should just fix the implementation in MCMCDiagnosticTools?

**Comment by sethaxen** [comment]
Agreed this should be fixed upstream. @torfjelde can you by any chance share an MWE?

**Comment by torfjelde** [comment]
I don't have an MWE atm, but the model is the SDE example from https://turinglang.org/v0.24/tutorials/10-bayesian-differential-equations/ that I ran _without_ adaptation :confused:

**Comment by sethaxen** [comment]
@torfjelde I just tried and wasn't able to reproduce this. Could you write your chains to a file and share them with me?

**Comment by sethaxen** [comment]
@torfjelde gentle reminder to send the problematic chains when you get the chance

**Comment by sethaxen** [comment]
@torfjelde _slightly_ less gentle reminder to send the problematic chains

**Comment by torfjelde** [comment]
Haha yeah sorry. Basically my computer with the thingy crashed and so I couldn't access it until after I got back from BayesComp, and I've yet to be able to reproduce the example:/

**Comment by sethaxen** [comment]
I'm not certain what the original issue is, but if it's the same as #447, then MCMCDiagnosticTools v0.3.10 contains a fix for this: https://github.com/TuringLang/MCMCDiagnosticTools.jl/pull/119
